include/wpe/gamepad.h

```diff
@@ -0,0 +1,285 @@
+/*
+ * If not stated otherwise in this file or this component's Licenses.txt file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 RDK Management
+ * Copyright 2021 Igalia, S.L.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if !defined(__WPE_H_INSIDE__) && !defined(WPE_COMPILATION)
+#error "Only <wpe/wpe.h> can be included directly."
+#endif
+
+#ifndef wpe_gamepad_h
+#define wpe_gamepad_h
+
+#if defined(WPE_COMPILATION)
+#include <wpe/export.h>
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Standard mapping
+// Refer https://www.w3.org/TR/gamepad/#gamepadbutton-interface
+enum StandardGamepadAxis {
+    GAMEPAD_AXIS_UNKNOWN = -1,
+    GAMEPAD_AXIS_LEFT_STICK_X,
+    GAMEPAD_AXIS_LEFT_STICK_Y,
+    GAMEPAD_AXIS_RIGHT_STICK_X,
+    GAMEPAD_AXIS_RIGHT_STICK_Y,
+    GAMEPAD_AXIS_COUNT,
+};
+
+enum StandardGamepadButton {
+    GAMEPAD_BUTTON_UNKNOWN = -1,
+    GAMEPAD_BUTTON_A,
+    GAMEPAD_BUTTON_B,
+    GAMEPAD_BUTTON_X,
+    GAMEPAD_BUTTON_Y,
+    GAMEPAD_BUTTON_LEFT_SHOULDER,
+    GAMEPAD_BUTTON_RIGHT_SHOULDER,
+    GAMEPAD_BUTTON_LEFT_TRIGGER,
+    GAMEPAD_BUTTON_RIGHT_TRIGGER,
+    GAMEPAD_BUTTON_SELECT,
+    GAMEPAD_BUTTON_START,
+    GAMEPAD_BUTTON_LEFT_STICK,
+    GAMEPAD_BUTTON_RIGHT_STICK,
+    GAMEPAD_BUTTON_D_PAD_UP,
+    GAMEPAD_BUTTON_D_PAD_DOWN,
+    GAMEPAD_BUTTON_D_PAD_LEFT,
+    GAMEPAD_BUTTON_D_PAD_RIGHT,
+    GAMEPAD_BUTTON_COUNT,
+};
+
+struct wpe_gamepad;
+struct wpe_gamepad_provider;
+
+/* app -> WebKit */
+struct wpe_gamepad_provider_client_interface {
```

Comment:
These structures are undocumented.

Also a longer documentation comment with an overview about how the API is used would be very good, but I won't insist on having it for merging 😉